improver_tests/acceptance/acceptance.py

```diff
@@ -102,20 +106,44 @@ def recreate_if_needed(output_path, kgo_path):
 
     Args:
         output_path (pathlib.Path): Path to output produced by test
-        kgo_path (pathlib.Path): Path to KGO file
+        kgo_path (pathlib.Path): Path to expected/original KGO file
 
     Returns:
-        None
+        bool: True if KGO was recreated
     """
     if not kgo_recreate():
-        return
+        return False
     if not kgo_path.is_absolute():
         raise IOError("KGO path is not absolute")
-    if kgo_root() not in kgo_path.parents:
-        raise IOError("Provided KGO path is not within KGO root directory")
-    kgo_path.parent.mkdir(exist_ok=True)
-    shutil.copyfile(str(output_path), str(kgo_path))
-    return
+    if not output_path.is_file():
+        raise IOError("Expected output file not created by running test")
+    kgo_root_dir = kgo_root()
+    recreate_dir = pathlib.Path(os.environ[RECREATE_DIR])
+    if kgo_root_dir not in kgo_path.parents:
+        raise IOError("KGO path for test is not within KGO root directory")
+    if not recreate_dir.is_absolute():
+        raise IOError("Recreate KGO path is not absolute")
+    print("Comparison found differences - recreating KGO for this test")
+    if kgo_path.exists():
+        print(f"Original KGO file is at {kgo_path}")
+    else:
+        print("Original KGO file does not exist")
+    kgo_relative = kgo_path.relative_to(kgo_root_dir)
+    recreate_path = recreate_dir / kgo_relative
+    recreate_path.parent.mkdir(exist_ok=True, parents=True)
+    if recreate_path.exists():
+        recreate_path.unlink()
+    shutil.copyfile(str(output_path), str(recreate_path))
+    print(f"Updated KGO file is at {recreate_path}")
+    if recreate_path.resolve() == kgo_path.resolve():
+        print("KGO updated in place, rerunning this test should now pass")
+    else:
+        print(f"Put the updated KGO file in {ACC_TEST_DIR} to make this"
```

Comment:
Did you mean `os.environ[ACC_TEST_DIR]`?

Reply:
Referring to the environment variable name here was intentional to explain where the path comes from. The full path where to place the file will be shown in `cp` command in the next line.